lib/component-utils/paramsModule.js

```diff
@@ -1,21 +1,21 @@
 function updateParams(oldVnode, vnode) {
   let key;
-  let cur;
-  let old;
   const elm = vnode.elm;
   let oldParams = oldVnode.data.params;
   let params = vnode.data.params;
   if (!oldParams && !params) return;
   if (oldParams === params) return;
   oldParams = oldParams || {};
   params = params || {};
-  for (key in params) {
-    cur = params[key];
-    old = oldParams[key];
-    if (old !== cur && elm.params) {
-      elm.params[key] = cur;
-    } else {
-      elm[key] = cur;
+  if (elm._setParams) {
+    elm._setParams(params);
+  } else {
```

Comment:
I forgot to mention this earlier, I don't think there should be a prop fallback. We should never surprise users. If users wanted to use props, they'd use props, params should never quietly behave like props in certain scenarios. We should straight up throw an error if users attempt to apply params to a non-Panel element.